components/browser/state/src/main/java/mozilla/components/browser/state/state/CustomTabSessionState.kt

```diff
@@ -12,16 +12,23 @@ import java.util.UUID
  * @property id the ID of this custom tab and session.
  * @property content the [ContentState] of this custom tab.
  * @property trackingProtection the [TrackingProtectionState] of this custom tab.
+ * @property config the [CustomTabConfig] used to create this custom tab.
  */
 data class CustomTabSessionState(
     override val id: String = UUID.randomUUID().toString(),
     override val content: ContentState,
-    override val trackingProtection: TrackingProtectionState = TrackingProtectionState()
+    override val trackingProtection: TrackingProtectionState = TrackingProtectionState(),
+    val config: CustomTabConfig
 ) : SessionState
 
-internal fun createCustomTab(url: String, id: String = UUID.randomUUID().toString()): CustomTabSessionState {
+internal fun createCustomTab(
+    url: String,
+    id: String = UUID.randomUUID().toString(),
+    config: CustomTabConfig = CustomTabConfig()
```

Comment:
Eventually we may want to enforce providing a config (I'm going to make that helper visible in a different PR) but for now that's good enough (and probably means we do not have to change all the tests!).